### PR TITLE
[LM-10471] Added Delete Dialogs and its Styles and Customizations

### PR DESCRIPTION
**Pull request by sidcr7-likeminds**
## Description

**Features**

- Added delete dialogs
- Added styles related to delete dialogs
- Added customisations related to delete dialogs 

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fix for breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseBottomSheetFragment.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = LmFeedReasonChooseBottomSheetFragmentBinding.inflate(layoutInflater)
+
+        return binding.root
+    }
+
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseBottomSheetFragment.kt:41]: no customization given?

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseListView.kt
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseListView.kt:1]: add comments to public functions

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedSelfDeleteDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    ): View {
+        binding = LmFeedDialogFragmentSelfDeleteBinding.inflate(layoutInflater)
+
+        binding.apply {
+            customizeDeleteDialog(alertDialogDelete)
+            return root
+        }
+    }
+
+    protected open fun customizeDeleteDialog(alertDialogDelete: LMFeedAlertDialogView) {
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedSelfDeleteDialogFragment.kt:78]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedAdminDeleteDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    ): View {
+        binding = LmFeedDialogFragmentAdminDeleteBinding.inflate(layoutInflater)
+
+        binding.apply {
+            customizeDeleteDialog(alertDialogDelete)
+            return root
+        }
+    }
+
+    protected open fun customizeDeleteDialog(alertDialogDelete: LMFeedAlertDialogView) {
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedAdminDeleteDialogFragment.kt:79]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedSelfDeleteDialogFragment.kt [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    protected open fun onDeleteAlertPositiveButtonClicked() {
+        selfDeleteDialogListener?.onEntityDeletedByAuthor(deletedExtras)
+        dismiss()
+    }
+
+    protected open fun onDeleteAlertNegativeButtonClicked() {
+        dismiss()
+    }
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedSelfDeleteDialogFragment.kt:137]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/style/LMFeedPostViewStyle.kt [hunk 5]
```diff
@@ -30,8 +30,10 @@ class LMFeedPostViewStyle private constructor(
     val postMediaViewStyle: LMFeedPostMediaViewStyle,
     //post footer style
     val postFooterViewStyle: LMFeedPostFooterViewStyle,
-    //post topics style
-    val postTopicsViewStyle: LMFeedPostTopicsViewStyle
+    //post topics chip group style
+    val postTopicsGroupStyle: LMFeedChipGroupStyle,
+    //post topics chip style
+    val postTopicChipsStyle: LMFeedChipStyle
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/style/LMFeedPostViewStyle.kt:36]: add these changes in styling comments as well

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/style/LMFeedReportFragmentViewStyle.kt [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+import android.text.TextUtils
+import androidx.annotation.ColorRes
+import com.likeminds.feed.android.core.R
+import com.likeminds.feed.android.core.ui.base.styles.*
+import com.likeminds.feed.android.core.ui.widgets.alertdialog.style.LMFeedAlertDialogStyle
+import com.likeminds.feed.android.core.ui.widgets.headerview.style.LMFeedHeaderViewStyle
+import com.likeminds.feed.android.core.utils.LMFeedViewStyle
+import com.likeminds.feed.android.core.utils.model.LMFeedPadding
+
+class LMFeedReportFragmentViewStyle private constructor(
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/style/LMFeedReportFragmentViewStyle.kt:13]: add styling comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportFragment.kt [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+        }
+    }
+
+    protected open fun customizeReportFragmentHeaderView(headerViewReport: LMFeedHeaderView) {
+        headerViewReport.apply {
+            setStyle(LMFeedStyleTransformer.reportFragmentViewStyle.headerViewStyle)
+
+            setTitleText(getString(R.string.lm_feed_report_abuse))
+        }
+    }
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportFragment.kt:113]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportFragment.kt [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+
+    private fun initListeners() {
+        binding.btnPostReport.setOnClickListener {
+            onReportSubmitted()
+        }
+    }
+
+    protected open fun onReportSubmitted() {
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportFragment.kt:169]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportSuccessDialogFragment.kt [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    ): View {
+        binding = LmFeedDialogFragmentReportSuccessBinding.inflate(layoutInflater)
+
+        binding.apply {
+            customizeReportSuccessDialog(alertDialogDelete)
+            return root
+        }
+    }
+
+    protected open fun customizeReportSuccessDialog(alertDialogDelete: LMFeedAlertDialogView) {
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportSuccessDialogFragment.kt:34]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportTagListView.kt
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/report/view/LMFeedReportTagListView.kt:1]: add comments to public function

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedChipGroupStyle.kt [hunk 10]
```diff
@@ -0,0 +1,66 @@
+package com.likeminds.feed.android.core.ui.base.styles
+
+import androidx.annotation.DimenRes
+import com.likeminds.feed.android.core.R
+import com.likeminds.feed.android.core.ui.base.views.LMFeedButton
+import com.likeminds.feed.android.core.ui.base.views.LMFeedChipGroup
+import com.likeminds.feed.android.core.utils.LMFeedViewStyle
+
+class LMFeedChipGroupStyle private constructor(
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedChipGroupStyle.kt:9]: add styling comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedChipStyle.kt [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+import android.content.res.ColorStateList
+import android.util.TypedValue
+import androidx.annotation.ColorRes
+import androidx.annotation.DimenRes
+import androidx.core.content.ContextCompat
+import com.likeminds.feed.android.core.R
+import com.likeminds.feed.android.core.ui.base.views.LMFeedChip
+import com.likeminds.feed.android.core.utils.LMFeedViewStyle
+
+class LMFeedChipStyle private constructor(
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedChipStyle.kt:12]: add styling comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedTextStyle.kt [hunk 12]
```diff
@@ -49,6 +49,7 @@ class LMFeedTextStyle private constructor(
     @ColorRes val backgroundColor: Int?,
     val textAlignment: Int?,
     @DimenRes val elevation: Int?,
+    @ColorRes val hintTextColor: Int?,
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/styles/LMFeedTextStyle.kt:52]: add these change in styling comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/views/LMFeedChipGroup.kt [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ChipGroup(context, attrs, defStyleAttr) {
+
+    //removes all the chips from the chip group
+    fun removeAllChips() {
+        this.removeAllViews()
+    }
+
+    fun addChip(
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/base/views/LMFeedChipGroup.kt:22]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/widgets/alertdialog/style/LMFeedAlertDialogStyle.kt [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+import android.graphics.Typeface
+import android.text.TextUtils
+import androidx.annotation.ColorRes
+import androidx.annotation.DimenRes
+import com.likeminds.feed.android.core.R
+import com.likeminds.feed.android.core.ui.base.styles.LMFeedEditTextStyle
+import com.likeminds.feed.android.core.ui.base.styles.LMFeedTextStyle
+import com.likeminds.feed.android.core.utils.LMFeedViewStyle
+
+class LMFeedAlertDialogStyle private constructor(
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/widgets/alertdialog/style/LMFeedAlertDialogStyle.kt:12]: add styling comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/widgets/alertdialog/view/LMFeedAlertDialogView.kt
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/ui/widgets/alertdialog/view/LMFeedAlertDialogView.kt:1]: add comments to public functions

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/utils/LMFeedStyleTransformer.kt [hunk 15]
```diff
@@ -27,4 +32,45 @@ object LMFeedStyleTransformer {
...
+                    .textAllCaps(true)
+                    .textColor(R.color.lm_feed_black_20)
+                    .textSize(R.dimen.lm_feed_text_small)
+                    .typeface(Typeface.NORMAL)
+                    .fontResource(R.font.lm_feed_roboto_medium)
+                    .build()
+            )
+            .alertBoxElevation(R.dimen.lm_feed_elevation_small)
+            .alertBoxCornerRadius(R.dimen.lm_feed_corner_radius_small)
+            .build()
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/utils/LMFeedStyleTransformer.kt:75]: why were are building style here.

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/utils/LMFeedStyleTransformer.kt (hunk 15 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/utils/LMFeedStyleTransformer.kt:75]: initially I was trying to use an alert dialog directly without creating a fragment, that's why I updated the style here only. changing it to a fragment style now

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseBottomSheetFragment.kt (hunk 1 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/delete/view/LMFeedReasonChooseBottomSheetFragment.kt:41]: the customization of this fragment is inside the view data binder so there is no customization to give here

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**